backend/src/routes/api/scene.js

```diff
@@ -27,10 +29,27 @@ router.get("/full/:sceneId", async (req, res) => {
 // Retrieve all scenes of a scenario
 router.get("/", async (req, res) => {
   const scenes = await retrieveSceneList(req.params.scenarioId);
-
   res.json(scenes);
 });
 
+// Get graph
+router.get("/graph", async (req, res) => {
+  const scenes = await retrieveSceneList(req.params.scenarioId);
+  let fullScenes = [];
+  
+  // Get FULL scene data - should really change this to run in parallel for better runtime
+  for await (const scene of scenes) {
+    const fullScene = await retrieveScene(scene._id);
+    fullScenes.push(fullScene);
+  }
```

Comment:
You can parallelize async requests using `Promise.all()` - might be a good interim feature to speed up this process (which I think is worth it for this). Basically, something like...

```
const promises = scenes.map((it) => retrieveScene(it._id));
const fullScenes = await Promise.all(promises);
```

However, this still means we will make `n` requests to the mongoDB which isn't too great. See below comment for future improvement.

In the future, it'll be better if you could actually just pull all the info needed from the scenes with one request to MongoDB. @nickykay98 @Harshal-D this could be made into an improvement task (something like: create method `retrieveSceneListDetailed()` in `sceneDao.js` to retrieve all scenes and their details by providing a scenario ID).